src/hooks/stores/utils.ts

```diff
@@ -0,0 +1,303 @@
+
+export function checkData(
+  data: any,
+  setError: React.Dispatch<React.SetStateAction<string | undefined>>,
+  setIncorrectNetwork: React.Dispatch<React.SetStateAction<boolean>>,
+): void {
+  if (data) {
+    setError(undefined);
+    setIncorrectNetwork(false);
+  }
+}
+
+export function checkNetwork(
+  incorrectNetwork: boolean,
+  setIncorrectNetwork:React.Dispatch<React.SetStateAction<boolean>>,
+): void {
+  if (incorrectNetwork) {
+    setIncorrectNetwork(false);
+  }
+}
+
+export function validate(
+  formData: any,
+  chainId: any,
+  transactionData: any,
+  accountData: any,
+  workspace: MinimalWorkspace | undefined,
+  currentChainId: number | undefined,
+  switchNetwork: any,
+  setIncorrectNetwork: React.Dispatch<React.SetStateAction<boolean>>,
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  validatorApi: ValidationApi,
+  grantContract: any,
+) {
+  if (!formData) return;
+  if (transactionData) return;
+  if (!accountData || !accountData.address) {
+    throw new Error('not connected to wallet');
+  }
+  if (!workspace) {
+    throw new Error('not connected to workspace');
+  }
+  if (!currentChainId) {
+    if (switchNetwork && chainId) { switchNetwork(chainId); }
+    setIncorrectNetwork(true);
+    setLoading(false);
+    return;
+  }
+  if (chainId !== currentChainId) {
+    if (switchNetwork && chainId) { switchNetwork(chainId); }
+    setIncorrectNetwork(true);
+    setLoading(false);
+    return;
+  }
+  if (!validatorApi) {
+    throw new Error('validatorApi or workspaceId is not defined');
+  }
+  if (
+    !grantContract
+      || grantContract.address
+        === '0x0000000000000000000000000000000000000000'
+      || !grantContract.signer
+      || !grantContract.provider
+  // eslint-disable-next-line no-empty
+  ) {
+    // eslint-disable-next-line no-useless-return
+    return;
+  }
+}
+
+export function validateArchive(
+  transactionData: any,
+  accountData: any,
+  currentChainId: number | undefined,
+  workspace: MinimalWorkspace | undefined,
+  validatorApi: ValidationApi,
+  grantContract: any,
+) {
+  if (transactionData) return;
+  if (!accountData || !accountData.address) {
+    throw new Error('not connected to wallet');
+  }
+  if (!currentChainId) {
+    throw new Error('not connected to valid network');
+  }
+  if (!workspace) {
+    throw new Error('not connected to workspace');
+  }
+  if (getSupportedChainIdFromWorkspace(workspace) !== currentChainId) {
+    throw new Error('connected to wrong network');
+  }
+  if (!validatorApi) {
+    throw new Error('validatorApi or workspaceId is not defined');
+  }
+  if (
+    !grantContract
+      || grantContract.address
+        === '0x0000000000000000000000000000000000000000'
+      || !grantContract.signer
+      || !grantContract.provider
+  // eslint-disable-next-line no-useless-return
+  ) return;
+}
+
+// eslint-disable-next-line consistent-return
+export async function createGrant(
+  formData: any,
```

Comment:
need type safety here

Reply:
this can just be its own hook, there's no need to pass so many "any" typed parameters -- the function just becomes cumbersome to use & understand then